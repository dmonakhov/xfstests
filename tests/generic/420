#! /bin/bash
# FS QA Test 420
#
# Regression test for  information leak for lio-fileio target
# BUG: If image  file is less than virtual blockdev then read() may return
#      unitilized data.
#
#-----------------------------------------------------------------------
# Copyright (c) 2017 Dmitry Monakhov <dmonakhov@openvz.org>
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15


_cleanup()
{
	cd /
	rm -f $tmp.*
	_liotgt_cleanup
	rm -rf $TEST_DIR/$$
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/liotarget

# remove previous $seqres.full before test
rm -f $seqres.full

# real QA test starts here

# Modify as appropriate.
_supported_fs generic
_supported_os Linux
_require_test
_require_liotarget

mkdir -p $TEST_DIR/$$ || _fail "Can not make test dir"

cfg_path=$(_liotgt_create_fileio  420-test.img $TEST_DIR/$$/420-test.img 128M)
dev=$(_liotgt_attach_target /backstores/fileio/420-test.img)

$XFS_IO_PROG -f -c "truncate 1M" $TEST_DIR/$$/420-test.img >> $seqres.full

$XFS_IO_PROG -c "pwrite -S 0xa0 -b 512 512 1k" -d $dev >>$seqres.full 2>&1 || \
    _fail "pwrite failed"

$XFS_IO_PROG -c "pwrite -S 0xab -b 512 2M 1k" -d $dev >>$seqres.full 2>&1 || \
    _fail "pwrite failed"

md5sum $dev | sed -e "s,$dev,loopdev,g"
# success, all done
status=0
exit
