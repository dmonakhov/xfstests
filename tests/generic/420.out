QA output created by 420
f5cfb0d8d7bfadbc2127f4f6ca4a0eba  loopdev
